src/Lecture2.hs

```diff
@@ -70,7 +82,9 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt a xs = if a >= length xs || a < 0 then (Nothing, xs) else (Just (xs !! a), let (as, bs) = splitAt a xs in as ++ tail bs)
```

Comment:
Unfortunately, `length` hangs on infinite lists so your implementation doesn't work with infinite lists ♾️ 

Also, you traverse the list prefix twice: with the `!!` operator and with `splitAt`. But it's possible to implement this function using only a single traversal.